src/main/webapp/app/common/domain/User.ts

```diff
@@ -1,10 +1,6 @@
-export class UserCredentialsDTO {
+export interface User {
```

Comment:
Either you create a Login interface with only Login required info and you will create a User interface later.
Either you do both but in this case, I think role and id field is associated to a User, and you have to create a Login interface.